code/modules/atmospherics/machinery/components/trinary_devices/hypertorus.dm

```diff
@@ -0,0 +1,893 @@
+			. = FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/corner))
+			var/dir = get_dir(src,object)
+			if(dir in GLOB.cardinals)
+				. =  FALSE
+			switch(dir)
+				if(SOUTHEAST)
+					if(object.dir != SOUTH)
+						. = FALSE
+				if(SOUTHWEST)
+					if(object.dir != WEST)
+						. =  FALSE
+				if(NORTHEAST)
+					if(object.dir != EAST)
+						. =  FALSE
+				if(NORTHWEST)
+					if(object.dir != NORTH)
+						. =  FALSE
+			corners |= object
+			continue
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/interface))
+			if(linked_interface && linked_interface != object)
+				. =  FALSE
+			linked_interface = object
+
+	for(var/obj/machinery/atmospherics/components/unary/hypertorus/object in orange(1,src))
+		if(. == FALSE)
+			break
+
+		if(!object.anchored)
+			. = FALSE
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/fuel_input))
+			if(linked_input && linked_input != object)
+				. =  FALSE
+			linked_input = object
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/waste_output))
+			if(linked_output && linked_output != object)
+				. =  FALSE
+			linked_output = object
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input))
+			if(linked_moderator && linked_moderator != object)
+				. =  FALSE
+			linked_moderator = object
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/activate()
+	if(!active)
+		message_admins("YES")
+		active = TRUE
+		linked_interface.active = TRUE
+		linked_input.active = TRUE
+		linked_output.active = TRUE
+		linked_moderator.active = TRUE
+		for(var/obj/machinery/hypertorus/corner/corner in corners)
+			corner.active = TRUE
+	else
+		message_admins("Already connected")
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/deactivate()
+	if(active)
+		message_admins("YES")
+		active = FALSE
+		if(linked_interface)
+			linked_interface.active = FALSE
+		if(linked_input)
+			linked_input.active = FALSE
+		if(linked_output)
+			linked_output.active = FALSE
+		if(linked_moderator)
+			linked_moderator.active = FALSE
+		if(corners.len)
+			for(var/obj/machinery/hypertorus/corner/corner in corners)
+				corner.active = FALSE
+	else
+		message_admins("Already connected")
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/check_fuel()
+	if(internal_fusion.gases[/datum/gas/tritium][MOLES] > FUSION_MOLE_THRESHOLD && internal_fusion.gases[/datum/gas/hydrogen][MOLES] > FUSION_MOLE_THRESHOLD)
+		return TRUE
+	return FALSE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/process()
+	if(COOLDOWN_FINISHED(src, hypertorus_reactor))
+		slowprocess()
+		COOLDOWN_START(src, hypertorus_reactor, 1 SECONDS) //Set to wait for another second before processing again, we don't need to process more than once a second
```

Comment:
You could use `SSprocessing` to get a process every second.

Reply:
how do i do that? also i may remove it and just do process() or even process_atmos() i'm using cooldowns for a better control for debug purposes